src/Stryker.Core/Stryker.Core/ProjectComponents/ProjectComponent.cs

```diff
@@ -5,15 +5,36 @@
 
 namespace Stryker.Core.ProjectComponents
 {
-    public interface IProjectComponent
+    public delegate void Display(IReadOnlyProjectComponent current);
+
+    public interface IReadOnlyProjectComponent
     {
-        string FullPath { get; set; }
-        IEnumerable<Mutant> Mutants { get; set; }
-        IFolderComposite Parent { get; set; }
+        string FullPath { get; }
+        IEnumerable<Mutant> Mutants { get; }
+        IReadOnlyFolderComposite Parent { get; }
         string RelativePath { get; set; }
+        public Display DisplayFile { get; set; }
+        public Display DisplayFolder { get; set; }
+        IEnumerable<IReadOnlyMutant> TotalMutants();
+        IEnumerable<IReadOnlyMutant> ValidMutants();
+        IEnumerable<IReadOnlyMutant> UndetectedMutants();
+        IEnumerable<IReadOnlyMutant> DetectedMutants();
+        IEnumerable<IReadOnlyMutant> InvalidMutants();
+        IEnumerable<IReadOnlyMutant> IgnoredMutants();
+        IEnumerable<IReadOnlyMutant> NotRunMutants();
```

Comment:
These read as properties but are methods, why?

Reply:
Sonar told me to do this. As they iterate lists they should not be properties